### PR TITLE
Strict children

### PR DESCRIPTION
**Pull request by sgrishchenko**
cc @turansky @aerialist7

### CODE REVIEW

File: kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt [hunk 1]
```diff
@@ -88,11 +88,11 @@ open class RDOMBuilderImpl<out T : Tag>(factory: (TagConsumer<Unit>) -> T) : RDO
         }
 
         override fun onTagContent(content: CharSequence) {
-            childList.add(content)
+            childList.add(ReactNode(content.toString()))
```
Comment (turansky) [inline kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt:91]: `+` ?

File: kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt [hunk 2]
```diff
@@ -88,11 +88,11 @@ open class RDOMBuilderImpl<out T : Tag>(factory: (TagConsumer<Unit>) -> T) : RDO
...
         }
 
         override fun onTagContentEntity(entity: Entities) {
-            childList.add(entity.text)
+            childList.add(ReactNode(entity.text))
```
Comment (turansky) [inline kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt:95]: `+` ?

File: kotlin-react/src/main/kotlin/react/ReactNode.kt [hunk 3]
```diff
@@ -12,6 +12,9 @@ fun ReactNode.asElementOrNull(): ReactElement? = when {
     else -> null
 }
 
+fun ReactNode(string: String): ReactNode =
```
Comment (turansky) [inline kotlin-react/src/main/kotlin/react/ReactNode.kt:15]: It must be removed after #603

**Review by turansky — CHANGES_REQUESTED** [review]

File: kotlin-react/src/main/kotlin/react/RBuilder.kt [hunk 4]
```diff
@@ -20,7 +20,7 @@ interface RBuilder {
     }
 
     operator fun String.unaryPlus() {
-        childList.add(this)
+        childList.add(ReactNode(this))
```
Comment (turansky) [inline kotlin-react/src/main/kotlin/react/RBuilder.kt:23]: `child(ReactNode(this))`?

File: kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt (hunk 1 above)
Comment (sgrishchenko) [reply kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt:91]: fixed

File: kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt (hunk 2 above)
Comment (sgrishchenko) [reply kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt:95]: fixed

File: kotlin-react/src/main/kotlin/react/RBuilder.kt (hunk 4 above)
Comment (sgrishchenko) [reply kotlin-react/src/main/kotlin/react/RBuilder.kt:23]: both `unaryPlus` operators are fixed

File: kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt [hunk 5]
```diff
@@ -88,11 +88,11 @@ open class RDOMBuilderImpl<out T : Tag>(factory: (TagConsumer<Unit>) -> T) : RDO
         }
 
         override fun onTagContent(content: CharSequence) {
-            childList.add(content)
+            +content.toString()
         }
 
         override fun onTagContentEntity(entity: Entities) {
-            childList.add(entity.text)
+            +ReactNode(entity.text)
```
Comment (turansky) [inline kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt:95]: `+entity.text`?

**Comment by Leonya** [comment]
Released in pre.229

**Pull request merged**